tests/unit/jest.config.js

```diff
@@ -1,20 +1,31 @@
 const usePreact = Boolean(process.env.PREACT && JSON.parse(process.env.PREACT));
-const preactModuleMapper = usePreact
-  ? {
-      '^react$': 'preact/compat',
-      '^react-dom/test-utils$': 'preact/test-utils',
-      '^react-dom$': 'preact/compat',
-      '^react/jsx-runtime$': 'preact/jsx-runtime',
-      '^@testing-library/react-hooks$': '@testing-library/preact-hooks',
-    }
-  : {};
+const useWebComponents = Boolean(process.env.WC && JSON.parse(process.env.WC));
+const preactModuleMapper =
+  usePreact || useWebComponents
+    ? {
+        '^react$': 'preact/compat',
+        '^react-dom/test-utils$': 'preact/test-utils',
+        '^react-dom$': 'preact/compat',
+        '^react/jsx-runtime$': 'preact/jsx-runtime',
+        '^@testing-library/react-hooks$': '@testing-library/preact-hooks',
+      }
+    : {};
+
+const matchWebComponentTests = useWebComponents && {
+  testMatch: ['<rootDir>/design-system/src/components/web-components/**/*.test.[jt]s(x)?'],
+};
+
+const ignoreWebComponentTests = !useWebComponents && {
+  testPathIgnorePatterns: ['<rootDir>/design-system/src/components/web-components'],
+};
```

Comment:
These could possibly just be combined into one object. The names are possibly confusing when you just look at the big config object below. Like does it mean that we're both "matching web component tests" and "ignoring web component tests"?

```suggestion
const conditionalWebComponentsConfig = useWebComponents ? {
  testMatch: ['<rootDir>/design-system/src/components/web-components/**/*.test.[jt]s(x)?'],
} : {
  testPathIgnorePatterns: ['<rootDir>/design-system/src/components/web-components'],
};
```

or something like this?

Reply:
works like a dream!